src/html-hooks.ts

```diff
@@ -33,39 +33,47 @@ export const useMediaQuery = (query: string): boolean => {
  */
 export const useScrolling = (scrollable: HTMLElement | undefined | null) => {
   const [scrolling, setScrolling] = React.useState(false);
-  let touchStartCount = 0;
 
   React.useEffect(() => {
+    let touching = false;
     let timer: NodeJS.Timeout | undefined;
-    const listener = () => {
+
+    const scrollHandler = () => {
       if (timer) {
         clearTimeout(timer);
       }
       setScrolling(true);
-      if (touchStartCount === 0) {
-        // If touchStartCount is 0, the user initiated scrolling using an attached pointing device (mouse or trackpad).
+      if (!touching) {
+        // If not touching, the user initiated scrolling using an attached pointing device (mouse or trackpad).
         timer = setTimeout(() => setScrolling(false), 250);
       }
     };
+
+    const touchStartHandler = () => {
+      touching = true;
+    };
+
+    const touchEndHandler = (ev: TouchEvent) => {
+      touching = ev.touches.length !== 0;
+      if (!touching) {
+        setScrolling(false);
+      }
+    };
+
     if (scrollable) {
-      scrollable.ontouchstart = () => {
-        ++touchStartCount;
-      };
-      scrollable.ontouchend = () => {
-        --touchStartCount;
-        if (touchStartCount === 0) {
-          setScrolling(false);
-        }
-      };
-      scrollable.addEventListener("scroll", listener);
+      scrollable.addEventListener("scroll", scrollHandler);
+      // these events are intentionally only for touch as drag scrolling isn't done with the mouse or trackpad
+      scrollable.addEventListener("touchstart", touchStartHandler);
+      scrollable.addEventListener("touchend", touchEndHandler);
     }
+
     return () => {
```

Comment:
why not move this return statement into the if above, so that you don't need to check `if (scrollable)` again?

Reply:
We always have to return an arrow function, so doing that would require an else clause returning an empty arrow function. I think in this case it's better to have a single return to make it more clear what the cleanup code always does.

Reply:
does `useEffect` not allow dynamic cleanup function return? I don't recall that being the case, I'll check

Reply:
We could do this if you think it's cleaner:

```tsx
    if (scrollable) {
      scrollable.addEventListener("scroll", scrollHandler);
      // these events are intentionally only for touch as drag scrolling isn't done with the mouse or trackpad
      scrollable.addEventListener("touchstart", touchStartHandler);
      scrollable.addEventListener("touchend", touchEndHandler);

      return () => {
        scrollable.removeEventListener("scroll", scrollHandler);
        scrollable.removeEventListener("touchstart", touchStartHandler);
        scrollable.removeEventListener("touchend", touchEndHandler);
      };
    }
    return () => { };
```

Reply:
while the first half is what I was proposing, you should not need the final `return () => {}`. I just checked without typescript React does not consider it an error to dynamically stop returning a cleanup function, it's fine to note return anything. I'm currently checking if typescript will be unhappy

Reply:
I confirmed typescript also has no issues with it. It should be completely fine to just do this:
```tsx
    if (scrollable) {
      scrollable.addEventListener("scroll", scrollHandler);
      // these events are intentionally only for touch as drag scrolling isn't done with the mouse or trackpad
      scrollable.addEventListener("touchstart", touchStartHandler);
      scrollable.addEventListener("touchend", touchEndHandler);

      return () => {
        scrollable.removeEventListener("scroll", scrollHandler);
        scrollable.removeEventListener("touchstart", touchStartHandler);
        scrollable.removeEventListener("touchend", touchEndHandler);
      };
    }
    ```

Reply:
basically letting the function return by reach its end, returning explicitly, or returning the undefined value explicitly all have the same behavior in react effects:
```ts
return () => {};
return undefined;
return;
```
will all have the same behavior of doing nothing during react's effect cleanup stage in component lifecycles

Reply:
I tried removing the no-op return (last line in above comment) and it compiles in this project, but the same code doesn't compile in our project. I think it's due to @bentley/build-tools/tsconfig-base.json specifying `"noImplicitReturns": true,`

Reply:
then you should be able to do either `return;` or `return undefined`, if you want to keep that rule. It depends on if `@types/react` types useEffect effects as returning `undefined | () => {}` or `void | () => {}`, I would have to check.

Reply:
it expects the result to be void if there is no callback, (the full expected return type is  `void |  (()=>void|undefined)`)
So if you want to keep that eslint rule you'll need to explicitly do `return;` instead of `return () => {}`

Reply:
I don't mean it to seem like I'm pressing on this. It's not a big issue, but I do think it'll be cleaner if you don't match the `if` in the cleanup, and its technically a micro-optimization since React never has to think about clean up if it knows you didn't provide it one (the optimization is negligible, I think the cleanliness is the important part).

I do think an explicit `return;` is pretty ugly, but I would value logical closeness of the cleanup to the setup in this case.

Reply:
I'm not positive, but I think that our eslint rule also won't allow us to return both a value and void from the same function. If that's the case, then I personally think that the Todd's original code is probably best.

Reply:
I've given my 2 cents, I'm pretty sure since the return type is a union of `void` and `()=>void` you can explicitly return void using `return;`, but whichever you prefer.

Reply:
actually I just realized, this eslint is probably setup in this repo. I somehow thought it was in your repo where you're sourcing the hook from. If it's in this repo, I'll disable that rule.

Reply:
It's not eslint that's producing an error when we don't have a return outside the if statement, it is the typescript compiler. Please see my comment above about using the standard bentley tsconfig-base. 

Having `return;` works instead of a no-op arrow function.

I'll push a change for this.

Reply:
oh, no-implicit-returns is a typescript compiler option... my bad. I still prefer the `return;` in that case. I didn't see that issue when testing on a local clone of this repo to check the types when optionally returning a clean up function, either this package's tsconfig doesn't have that on or something on my end is broken. Even if we publish without `return;` it shouldn't be a problem for consumers of this package since the return is an implementation detail and not part of any exported types.

Reply:
confirmed it is not on for this package, and none of the meta-options for tsc contain it

Reply:
Actually, since this package doesn't have that tsconfig requirement, I just moved the cleanup return into the if block and we don't need an empty return statement.